VE-APIs/src/controllers/user.ts

```diff
@@ -0,0 +1,103 @@
+import express, { Request, Response } from "express";
+import mongoose from "mongoose";
+import { IUser, User } from "../models/user";
+import Joi from "joi";
+
+export function generateRandomCodes(
+  Codecount: number
+): Array<{ gameCode: string }> {
+  const codes: Array<{ gameCode: string }> = [];
+  const characters_Upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const numbers = "0123456789";
+  const characters_Lower = "abcdefghijklmnopqrstuvwxyz";
+
+  const dictionary = characters_Lower + numbers + characters_Upper;
+
+  for (let i = 0; i < Codecount; i++) {
+    const codeLength = 8;
+    let code = "";
+
+    for (let j = 0; j < codeLength; j++) {
+      const randomIndex = Math.floor(Math.random() * dictionary.length);
+      code += dictionary.charAt(randomIndex);
+    }
+    codes.push({ gameCode: code });
+  }
+  return codes;
+}
+
+function validateNoOfGameCodes(code: number) {
+  const schema = Joi.object({
+    noOfGameCodes: Joi.number().integer().min(1).max(100).required(),
+  });
+
+  return schema.validate(code);
+}
+
+function validateGameCode(gamecode: { GameCode: string }) {
+  const schema = Joi.object({
+    GameCode: Joi.string().min(8).max(8).required(),
+  });
+
+  return schema.validate(gamecode);
+}
+
+export const healthcheck = async (req: Request, res: Response) => {
+  res.status(200).json({ message: "Server is up and running" });
+};
+export const generateGameCodes = async (req: Request, res: Response) => {
+  //it shall recive no of game codes to be generated from request
+  const { error, value } = validateNoOfGameCodes(req.body);
+  if (error) {
+    console.log(`Error is : ${error}`);
+    return res.status(400).json({ message: error.details[0].message });
+  }
+
+  const numberOfGameCode: number = value.noOfGameCodes;
+  const randomGameCodes = generateRandomCodes(numberOfGameCode);
+  try {
+    const savedCodes = await User.insertMany(randomGameCodes);
+    console.log(`Following are the generated codes : 
+    ${savedCodes}`);
+    res.status(200).json(savedCodes);
+  } catch (error) {
+    console.log(`Error is : ${error}`);
+    res.status(500).json({ message: "Failed to generate game codes" });
+  }
+};
+
+export const login = async (req: Request, res: Response) => {
+  const { error } = validateGameCode({ GameCode: req.body.gameCode });
+  if (error) {
+    console.log(`Error is : ${error}`);
+    return res.status(400).json({ message: error.details[0].message });
+  }
+  const GameCode = req.body.gameCode;
+
+  try {
+    const user = await User.findOne({ gameCode: GameCode });
```

Comment:
we are hitting to DB, if a robot hit our API multiple times then it will shutdown our DB so find out some package which will block the user on the base of IP

Reply:
There is some plugin by name limiter